bool ok
int32 point_x
float32 point_z
